### DISCUSSION TITLE
Download a time series of Sentinel 2 images without getting an error message

### DISCUSSION THREAD

**Discussion by Diieye** [microsoft/PlanetaryComputer#315] [General]
Hello,
I want to download a time series of Sentinel-2 images over one year using the command line (moringa fetch [-h] [--date_range DATE_RANGE] [--auth AUTH] imagery roi out_folder

positional arguments:
  imagery               Name of the images to fetch (supp.: s2theia, s2planetary, s1grd, s1rtc, planetmosaics,
                        nasadem, cop-dem-glo-30.
  roi                   Path to a vector GIS file whose extent will be used for spatial query.
  out_folder            Output folder where fetched data will be downloaded.

optional arguments:
  -h, --help            show this help message and exit
  --date_range DATE_RANGE
                        Date query in the YYYY-MM-DD/YYYY-MM-DD format. (default: None)
  --auth AUTH           Authentication information (credentials file, API key, etc.) (default: None)
).

However, I frequently encounter errors during download, with a message like: "rasterio.errors.RasterioIOError: Read or write failed. IReadBlock
failed at X offset 8, Y offset 12: TIFFReadEncodedTile() failed.." As a result, the download is interrupted, and I have to restart it. This often leads to image folders containing missing files after the download. Thus, when starting image preprocessing, error messages indicating missing files persist.

It was suggested that I use an API key for authentication, but this didn't solve the problem significantly.

My question is: how can I download time series without encountering these errors?

error message when downloading time series  :   ![image_2024_01_30T13_14_54_991Z](https://github.com/microsoft/PlanetaryComputer/assets/103746482/4ddc02eb-1511-4a16-9c7a-6dc4e20a5ddb)

error message preprocessing :  
![Error preprocessing](https://github.com/microsoft/PlanetaryComputer/assets/103746482/615120e2-1481-4e3e-888c-0c104361194c)

Thank you

**Comment by TomAugspurger** [comment]
Maybe a basic question first: do you need to download the data to disk? We typically recommend users stream data directly from Blob Storage into their application.

Aside from that, you probably want to set `GDAL_HTTP_MAX_RETRY` in the process making requests to Blob Storage. https://gdal.org/user/configoptions.html
